js/hoverzoom.js

```diff
@@ -3682,7 +3682,12 @@ var hoverZoom = {
         }
 
         function saveImage() {
-            cLog('check for media saving option');
+            cLog('check for permission');
+            chrome.runtime.sendMessage({action:'getPermissionsContains', permissions: ['downloads']}, 
+                function(response) {
+                    options.allowMediaSaving = response;
+            });
+                
             if (options.allowMediaSaving) {
```

Comment:
The problem is that this is asynchronous call. The nested response handler will be executed _after_ the rest of `saveImage()` function completes. You need to move entire body of the function inside the response handler.